fyle_accounting_library/common_resources/models.py

```diff
@@ -0,0 +1,75 @@
+from django.db import models
+
+from apps.workspaces.models import Workspace
+
+from .helpers import generate_choices_from_enum
+from .enums import DimensionDetailSourceTypeEnum
+
+
+SOURCE_TYPE_CHOICES = generate_choices_from_enum(DimensionDetailSourceTypeEnum)
+
+
+class DimensionDetail(models.Model):
+    """
+    Sage Intacct Dimension Details
+    DB Table: dimension_details:
+    """
+    id = models.AutoField(primary_key=True)
+    workspace = models.ForeignKey(Workspace, on_delete=models.PROTECT, help_text='Reference to Workspace')
+    attribute_type = models.CharField(max_length=255, help_text='Sage Intacct Attribute Type')
+    display_name = models.CharField(max_length=255, help_text='Sage Intacct Attribute Term Label')
+    source_type = models.CharField(help_text='Source Type', null=False, default=DimensionDetailSourceTypeEnum.ACCOUNTING.value, max_length=100, choices=SOURCE_TYPE_CHOICES)
+    created_at = models.DateTimeField(auto_now_add=True, help_text='Created at')
+    updated_at = models.DateTimeField(auto_now=True, help_text='Updated at')
+
+    class Meta:
+        db_table = 'dimension_details'
+        unique_together = ('attribute_type', 'workspace_id', 'source_type')
```

Comment:
let's also add display name to this - any update in display name shouldn't cause accidental dups